span.go

```diff
@@ -29,6 +29,7 @@ import (
 	"github.com/opentracing/opentracing-go/ext"
 	"github.com/opentracing/opentracing-go/log"
 
+	j "github.com/uber/jaeger-client-go/thrift-gen/jaeger"
```

Comment:
I would prefer not to leak thrift model into domain model. Why not just define a local type and only convert it in the thrift converter?

The field `span.references` can be just `[]Reference`, not `[]*Reference`, to minimize memory allocations.